File: samples/C# Samples/ConsoleAcquireImageSample/ConsoleAcquireImageSample/TwainObject.cs
using System;
using System.Collections.Generic;
using System.Text;
using Dynamsoft.TWAIN;
using Dynamsoft.TWAIN.Interface;
using System.Threading;
using System.IO;
using System.Runtime.InteropServices;

namespace ConsoleAcquireImageSample
{
    public partial class TwainObject:Dynamsoft.TWAIN.Interface.IAcquireCallback
    {
        private TwainManager m_TwainManager = null;
        short iPageNumber = 0;
        private string m_StrProductKey = "t0068MgAAAENENwNWc7+efmkY+t7se6XaRPFZkvfB7QWiTjHiLykxngQdY09pzVtOvrefXBbVvYFbJSluECHlyxaOvHwUADk=";
        public TwainObject()
        {
            m_TwainManager = new TwainManager(m_StrProductKey);
        }


        public bool SelectSource()
        {
            bool bRet = false;
            Console.WriteLine("Select a scanner:");
            short sSourceCount = m_TwainManager.SourceCount;
            for (short sIndex = 0; sIndex < sSourceCount; sIndex++)
            {
                Console.WriteLine(sIndex.ToString() + ". "+m_TwainManager.SourceNameItems(sIndex));
            }
            Console.WriteLine("Enter the scanner index:");
            string temp = Console.ReadLine();
            int res = Int32.Parse(temp);
            if (res > -1 && res < m_TwainManager.SourceCount)
            {
                 bRet = m_TwainManager.SelectSourceByIndex((short)res);
                Console.WriteLine("Current Source name: " + m_TwainManager.CurrentSourceName);
            }
            if (!bRet)
            {
                Console.WriteLine("Fail to select source.");
            }
            return bRet;
        }

        public void AcquireImage()
        {
            m_TwainManager.IfDisableSourceAfterAcquire = true;
           m_TwainManager.AcquireImage(this as IAcquireCallback);
        }

        public void Dispose()
        {
            m_TwainManager.Dispose();
        }
        #region IAcquireCallback Members

        public void OnPostAllTransfers()
        {
            m_TwainManager.CloseSourceManager();
            Console.WriteLine("Press any key to exit...");
            Console.ReadKey(true);
            Environment.Exit(0);
        }

        public bool OnPostTransfer(System.Drawing.Bitmap bit)
        {
  
            Console.WriteLine("The "+iPageNumber.ToString() +" pages of document has been scanned.");
            string path = System.IO.Path.GetTempPath();
            DateTime temp = DateTime.Now;
            int year = temp.Year;
            int month = temp.Month;
            int day = temp.Day;
            int hour = temp.Hour;
            int min = temp.Minute;
            int sec = temp.Second;
            string date = "Page" + iPageNumber.ToString()+"_" + year.ToString() + "_" + month.ToString() + "_" + day.ToString() + "_" + hour.ToString()+"hour" + "_"+min.ToString()+"min" + "_" + sec.ToString() + "sec";

            string filePath = path + date.ToString() + ".bmp"; 
            bit.Save(filePath);
            Console.WriteLine("Image has been saved. "+"The file path: "+ filePath.ToString());
            return true;

        }

        public void OnPreAllTransfers()
        {
            Console.WriteLine("The scan task will begin.");
        }

        public bool OnPreTransfer()
        {
            iPageNumber++;
            Console.WriteLine("The " + iPageNumber.ToString() + " pages of document will be scanned.");
            return true;
        }

        public void OnSourceUIClose()
        {
            
        }

        public void OnTransferCancelled()
        {

        }

        public void OnTransferError()
        {
        }

        #endregion
    }
}

File: samples/VB .NET Samples/ConsoleAcquireImageSample/ConsoleAcquireImageSample/TwainObject.vb
Imports System.Collections.Generic
Imports System.Text
Imports Dynamsoft.TWAIN
Imports Dynamsoft.TWAIN.Interface
Imports System.IO
Imports System.Runtime.InteropServices

Public Class TwainObject
    Implements Dynamsoft.TWAIN.Interface.IAcquireCallback
    Private m_TwainManager As TwainManager = Nothing
    Private iPageNumber As Short = 0
    Private m_StrProductKey As String
    Public Sub New()
        m_StrProductKey = "t0068UwAAAEQABDxqjGfgEzhVYureL0kGxugcsvIqCDGTPTsR5nLaQsNupIc17Y5vpMZAWBDsd6Xw3NMYzdHlHwiKUrfe/cU="
        m_TwainManager = New TwainManager(m_StrProductKey)
    End Sub


    Public Function SelectSource() As Boolean
        Dim bRet As Boolean = False
        Console.WriteLine("Select a scanner:")
        Dim sSourceCount As Short = m_TwainManager.SourceCount
        For sIndex As Short = 0 To sSourceCount - 1
            Console.WriteLine(sIndex.ToString() + ". " + m_TwainManager.SourceNameItems(sIndex))
        Next
        Console.WriteLine("Enter the scanner index:")
        Dim temp As String = Console.ReadLine()
        Dim res As Integer = Int32.Parse(temp)
        If res > -1 AndAlso res < m_TwainManager.SourceCount Then
            bRet = m_TwainManager.SelectSourceByIndex(CShort(res))
            Console.WriteLine("Current Source name: " + m_TwainManager.CurrentSourceName)
        End If
        If Not bRet Then
            Console.WriteLine("Fail to select source.")
        End If
        Return bRet
    End Function

    Public Sub AcquireImage()
        m_TwainManager.IfDisableSourceAfterAcquire = True
        m_TwainManager.AcquireImage(TryCast(Me, IAcquireCallback))
    End Sub

    Public Sub Dispose()
        m_TwainManager.Dispose()
    End Sub
#Region "IAcquireCallback Members"

    Public Sub OnPostAllTransfers() Implements IAcquireCallback.OnPostAllTransfers
        m_TwainManager.CloseSourceManager()
        Console.WriteLine("Press any key to exit...")
        Console.ReadKey(True)
        Environment.[Exit](0)
    End Sub

    Public Function OnPostTransfer(bit As System.Drawing.Bitmap) As Boolean Implements IAcquireCallback.OnPostTransfer

        Console.WriteLine("The " + iPageNumber.ToString() + " pages of document has been scanned.")
        Dim path As String = System.IO.Path.GetTempPath()
        Dim temp As DateTime = DateTime.Now
        Dim year As Integer = temp.Year
        Dim month As Integer = temp.Month
        Dim day As Integer = temp.Day
        Dim hour As Integer = temp.Hour
        Dim min As Integer = temp.Minute
        Dim [date] As String = year.ToString() + "_" + month.ToString() + "_" + day.ToString() + "_" + hour.ToString() + "hour" + "_" + min.ToString() + "min"

        Dim filePath As String = (path & Convert.ToString("\")) + [date].ToString() + ".bmp"
        bit.Save(filePath)
        Console.WriteLine("Image has been saved. " + "The file path: " + filePath.ToString())
        Return True

    End Function

    Public Sub OnPreAllTransfers() Implements IAcquireCallback.OnPreAllTransfers
        Console.WriteLine("The scan task will begin.")
    End Sub

    Public Function OnPreTransfer() As Boolean Implements IAcquireCallback.OnPreTransfer
        iPageNumber += 1
        Console.WriteLine("The " & iPageNumber.ToString() & " pages of document will be scanned.")
        Return True
    End Function

    Public Sub OnSourceUIClose() Implements IAcquireCallback.OnSourceUIClose

    End Sub

    Public Sub OnTransferCancelled() Implements IAcquireCallback.OnTransferCancelled

    End Sub

    Public Sub OnTransferError() Implements IAcquireCallback.OnTransferError
    End Sub

#End Region
End Class
